providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -85,23 +91,89 @@ public SharedKeyLiteAuthentication(SignatureWire signatureWire,
       this.crypto = crypto;
       this.utils = utils;
       this.signatureWire = signatureWire;
+	  this.storageUrl = URI.create("https://" + creds.get().identity + ".blob.core.windows.net/");
       this.creds = creds;
+	  this.identity = creds.get().identity;
+	  this.credential = creds.get().credential;
       this.timeStampProvider = timeStampProvider;
+	  this.sasAuthentication = false;
    }
-
+   
+   /** 
+   * this is an updated filter method, which decides whether the SAS or SharedKeyLite 
+   * is used and applies the right filtering.  
+   * 
+   */
    public HttpRequest filter(HttpRequest request) throws HttpException {
+      this.sasAuthentication = authSAS(this.credential);
+      if (this.sasAuthentication){
+         request = filterSAS(request, this.credential);
+      } else {
+         request = filterKey(request);
+      }
+      utils.logRequest(signatureLog, request, "<<");
+      return request;
+   }
+   
+   /**
+   * 
+   * this method checks the length of the authentication string, and decides, which auth method it represents. 
+   * 
+   */
+   public boolean authSAS(String credential){
+      int length = credential.length();
+      return  length < 150 && length > 120;    
+   }
+   
+   /** 
+   * this filter method is applied only for the cases with SAS Authentication. 
+   * 
+   */
+   public HttpRequest filterSAS(HttpRequest request, String credential) throws HttpException {
+      String containerName = null;
+      String blobName = null;
+      URI requestUri = request.getEndpoint();
+      try {
+         String[] parametersArray = cutUri(requestUri); 
+         containerName = parametersArray[1];
+         if (parametersArray.length == 3) {
+            blobName = parametersArray[2];
+            return removeAuthorizationHeader(replaceDateHeader(request.toBuilder().endpoint(Uris.uriBuilder(storageUrl).appendPath(containerName).appendPath(blobName).query(credential).build()).build()));
+         } 
+         return removeAuthorizationHeader(replaceDateHeader(request.toBuilder().endpoint(Uris.uriBuilder(storageUrl).appendPath(containerName).query("restype=container&" + credential).build()).addHeader("x-ms-version", "2018-03-28").build()));
+      } catch (NullPointerException e){
+         e.printStackTrace();
```

Comment:
Use a logger to log errors instead of printing to stdout

Reply:
Done!